### PR TITLE
stylelint: init at 16.2.0

### PR DESCRIPTION
**Pull request by eownerdead**
## Description of changes

Already, there is stylelint as a dependency of coc-stylelint.
nixpkgs-review fails with `nixpkgs_review.errors.NixpkgsReviewError: Failed to list packages: nix-env failed with exit code -9`.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by eownerdead** [comment]
Oh, I need to wait until #276732 is merged.

File: pkgs/by-name/st/stylelint/extensions/stylelint-config-clean-order/package.nix
Comment (applePrincess) [inline pkgs/by-name/st/stylelint/extensions/stylelint-config-clean-order/package.nix:19]: description sholu not end with period(.)

File: pkgs/by-name/st/stylelint/package.nix
Comment (applePrincess) [inline pkgs/by-name/st/stylelint/package.nix:35]: Description should not end with a period

**Comment by gepbird** [comment]
The current stylelint package only works for css files and fails for other dialects such as scss:
```
$ stylelint my-project/src/main.scss
Error: Cannot resolve custom syntax module "postcss-scss". Check that module "postcss-scss" is available and spelled correctly.

Caused by: Error [ERR_MODULE_NOT_FOUND]: Cannot find package 'postcss-scss' imported from /nix/store/wlnjf25g7w85z29q3x04ln304yp70yrz-stylelint-16.2.0/lib/node_modules
/stylelint/lib/utils/dynamicImport.mjs
```

This is because stylint marked dependencies like postcss-scss [dev dependencies](https://github.com/stylelint/stylelint/blob/16.2.0/package.json#L239) and buildNpmPackage doesn't install dev dependencies (at least by default, maybe there's a flag?).

Applying a [patch](https://github.com/stylelint/stylelint/commit/ab4d0a42df2bdc6a95675cd341f8c511543420d9.patch) that changes this dev-dependency to a normal dependency fixes the issue. Patching package.json and package-lock.json feels dirty and there must be a better way to do it with nix, this is just a quick solution I found.

I think we should add support for these css dialects either by adding it to the main stylelint package or making another package like stylelintFull.

I understand if this issue is out of the scope of this PR, just wanted to mention here if you are working on it anyway.

**Comment by eownerdead** [comment]
@gepbird We can add additional dependency by passing list to `withExtensions`.
So, probably `postcss-scss` can add like this.
Sorry for the very late response.

**Review by applePrincess — APPROVED** [review]
LGTM

**Comment by applePrincess** [comment]
Result of `nixpkgs-review pr 285348` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>stylelint</li>
  </ul>
</details>

**Comment by eownerdead** [comment]
Seems like there was a [commit](https://github.com/NixOS/nixpkgs/commit/db661a558ae858e64b2b6c3638ac1e7b2f343edd) similar to this one!

**Comment by katexochen** [comment]
Closing as implemented. :)
(Feel free to re-open if you think that's not correct.)

**Pull request closed without merging**
